mysticeti-core/src/synchronizer.rs

```diff
@@ -96,27 +73,53 @@ where
 
     pub async fn send_blocks(
         &mut self,
-        peer: AuthorityIndex,
+        peer: &Authority,
         references: Vec<BlockReference>,
     ) -> Option<()> {
+        const CHUNK_SIZE: usize = 10;
+
         let mut missing = Vec::new();
+        let mut to_send = vec![];
         for reference in references {
             let stored_block = self.inner.block_store.get_block(reference);
             let found = stored_block.is_some();
+
             match stored_block {
-                // TODO: Should we be able to send more than one block in a single network message?
-                Some(block) => self.sender.send(NetworkMessage::Block(block)).await.ok()?,
+                Some(block) => to_send.push(block),
                 None => missing.push(reference),
             }
+
+            if to_send.len() >= CHUNK_SIZE {
+                self.send(peer, NetworkMessage::Blocks(std::mem::take(&mut to_send)))?;
+            }
```

Comment:
If we do these kind of changes are we running benchmarking to ensure we are not regressing?

Reply:
You mean in AWS right? Basically those changes are introduced as part of our integration and experimentation in private-testnet where we already see better results and more sustainable behaviour. Also we are willing to run AWS experiments too.